gdal_subsetter/transform.py

```diff
@@ -668,23 +668,25 @@ def _regrid(infile, outfile, resampling_mode='bilinear', ref_crs=None, ref_box=N
     def recolor(self, layerid, srcfile, dstdir):
         fmt = self.message.format
         dstfile = srcfile  # passthrough if no colormap
-        # colormap = fmt.colormap
-
-        # Use hard coded colormaps that match layerid
-        colormaps_dir = os.path.dirname(os.path.realpath(__file__)) + '/colormaps/'
-        if 'ECCO' in srcfile:
-            colormap = colormaps_dir + 'MedspirationIndexed.txt'
-            discrete = True
-        elif 'analysed_sst' in srcfile:
-            colormap = colormaps_dir + 'GHRSST_Sea_Surface_Temperature.txt'
-            discrete = True
-        elif 'sst_anomaly' in srcfile:
-            colormap = colormaps_dir + 'GHRSST_Sea_Surface_Temperature_Anomalies.txt'
-            discrete = True
-        elif 'sea_ice_fraction' in srcfile:
-            colormap = colormaps_dir + 'GHRSST_Sea_Ice_Concentration.txt'
-            discrete = True
-        else:
+        colormap = None
+        variable = layerid.split('__')[1]
+
+        # Find the colormap URL from relatedUrls in matching variable
+        if '__all' not in layerid:  # '__all' tells us that there are no variables
+            variables = self.message.sources[0].variables
+            for var in variables:
```

Comment:
The level of cyclomatic complexity on this block of code feels a little hefty. Maybe it could be simplified a little with something like `next`. For example:

```
message_variable = next((variable for variables in self.message.source[0].variables
                         if var.name == variable), None)

if message_variable is not None:
     ...
```

There would be other changes following on from something like that.

Reply:
I was trying this out earlier but seemed harder to read for me.